Do not send HF token to other domains via `/proxy`

abidlabs opened the pull request:
As pointed out by @akx it's not secure to send the HF token to other domains via the `/proxy` route. This PR makes a change so that the HF token is only sent to HF domains

akx commented on gradio/routes.py:
```diff
@@ -144,6 +144,20 @@ def get_blocks(self) -> gradio.Blocks:
             raise ValueError("No Blocks has been configured for this app.")
         return self.blocks
 
+    @staticmethod
```
This could just as well not be a `@staticmethod` so it's easy to override in a subclass..?

akx replied:
To have an application be able to customize a forwarded request without having to copy-paste as much code.

abidlabs replied:
Hmm I'm having a hard time seeing this in practice -- nor have I ever seen the `App` class subclassed in any gradio app. If you have a concrete use case, feel free to share, otherwise I'll keep as is

freddyaboulton commented on gradio/routes.py:
```diff
@@ -144,6 +144,20 @@ def get_blocks(self) -> gradio.Blocks:
+        url = httpx.URL(url_path)
+        is_hf_url = (
+            url.host.endswith(".huggingface.co")
+            or url.host.endswith(".hf.space")
```
Since we just need this to fetch files from upstream spaces, we just need `.hf.space`, right?

abidlabs replied:
I think it should be fine to limit to just `.hf.space` yes, I'll make the change

freddyaboulton reviewed (approved):
Thank you for the fix @abidlabs !

abidlabs commented:
Addressed the suggestion and added a warning to the docstring about providing HF tokens only when loading trusted Spaces. Will merge after CI is green
